### ISSUE TITLE
📎 Support for cross language lint rules (`noUnusedVariables`, `noUnusedImports`, etc)

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#8590]
### Description

Right now, users are unable to use these very valuable rules because they currently don't function across language boundaries:
- `noUnusedVariables`
- `noUnusedImports`
- `useImportType`
- `useConst`

All of these rules rely on the semantic model to feed them the correct info. However, its currently confined to just javascript, and is unable to inspect the contents of the parent HTML-ish language to continue building the semantic model.

**Comment by ematipico** [comment]
What kind of discussion are we looking for?

**Comment by dyc3** [comment]
I mostly just want to collect info on the task here. It's kinda spread out in different places and in our heads, atm.

I remember us talking about adding an additional(?) service to support this (can't find the convo now). But if you've already assigned yourself, it seems like you already have a pretty good idea of how this is going to work :)

**Comment by ematipico** [comment]
Yeah I already have a working branch with `noUnusedVariables`.

The code is quite raw, but I'm getting there. There are few things I need to change in the svelte parser

**Comment by ematipico** [comment]
@dyc3 

What's a use case we need to fix for `useImportType`?

**Comment by dyc3** [comment]
There's a couple.

In this example, the rule can't detect the imported enum is used as a value in the markup, and suggests to turn it into a type import, which breaks the component.
```svelte
<script>
import FooEnum from "./helpers.ts"; // FooEnum is a typescript enum

let status: FooEnum | undefined = $state(undefined);
</script>

{#if status == FooEnum.Foo}
...
{/if}
```

A pattern that's common in `shadcn-svelte`:
```svelte
<script lang="ts">
import { Command as CommandPrimitive } from "bits-ui";
import { cn } from "$lib/utils.js";

let {
	ref = $bindable(null),
	class: className,
	...restProps
}: CommandPrimitive.LinkItemProps = $props(); // <-- used as type here
</script>

<!-- Can't detect used as value here -->
<CommandPrimitive.LinkItem
	class={cn(
		"...",
		className,
	)}
	bind:ref
	{...restProps}
/>
```

**Comment by ematipico** [comment]
Thank you. It seems that the current infrastructure can't support the rule yet, so I will need to add more data to the services

**Comment by rory-orennia** [comment]
Looks like it doesn't support params defined in the snippet function in Svelte for noUndeclaredVariables: [Playground](https://biomejs.dev/playground/?indentScriptAndStyle=true&lintRules=all&tab=formatter&pane=Diagnostics&code=ewAjAHMAbgBpAHAAcABlAHQAIABmAGkAZwB1AHIAZQAoAGkAbQBhAGcAZQApAH0ACgAJADwAZgBpAGcAdQByAGUAPgAKAAkACQA8AGkAbQBnACAAcwByAGMAPQB7AGkAbQBhAGcAZQAuAHMAcgBjAH0AIABhAGwAdAA9AHsAaQBtAGEAZwBlAC4AYwBhAHAAdABpAG8AbgB9ACAAdwBpAGQAdABoAD0AewBpAG0AYQBnAGUALgB3AGkAZAB0AGgAfQAgAGgAZQBpAGcAaAB0AD0AewBpAG0AYQBnAGUALgBoAGUAaQBnAGgAdAB9ACAALwA%2BAAoACQAJADwAZgBpAGcAYwBhAHAAdABpAG8AbgA%2BAHsAaQBtAGEAZwBlAC4AYwBhAHAAdABpAG8AbgB9ADwALwBmAGkAZwBjAGEAcAB0AGkAbwBuAD4ACgAJADwALwBmAGkAZwB1AHIAZQA%2BAAoAewAvAHMAbgBpAHAAcABlAHQAfQA%3D&language=svelte)

https://svelte.dev/docs/svelte/snippet

edit: Turns out it also doesn't work with noUnusedVariables
<img width="604" height="51" alt="Image" src="https://github.com/user-attachments/assets/19868e95-bb6f-4bb8-a287-2159bb0a836c" />
<img width="325" height="91" alt="Image" src="https://github.com/user-attachments/assets/fd473fb1-dea1-45df-b3ce-f42c4b8f4c3e" />

**Comment by dyc3** [comment]
@rory-orennia would you mind filing a separate issue for that?

**Comment by rory-orennia** [comment]
@dyc3 sure thing: https://github.com/biomejs/biome/issues/9653
I thought the blog for 2.4 said to report here so that's what I did :)
